docs/src/examples/SectionList.tsx

```diff
@@ -0,0 +1,288 @@
+      tableOfContentsRef.current?.scrollToIndex({
+        index: newIndex,
+        animated: true,
+      });
+    }
+    visibleIndex.value = newIndex;
+  };
+
+  const renderMeasuredItem: ListRenderItem<Section<T>> = ({ item, index }) => {
+    return (
+      <View
+        onLayout={({ nativeEvent }) => {
+          heights[index] = nativeEvent.layout.height;
+        }}>
+        <Text style={sectionListStyles.sectionHeader}> {item.name}</Text>
+        {renderItem(item.data)}
+      </View>
+    );
+  };
+
+  return (
+    <View style={sectionListStyles.flex}>
+      <FlashList
+        ref={contentListRef}
+        estimatedItemSize={100}
+        estimatedFirstItemOffset={0}
+        renderItem={renderMeasuredItem}
+        data={sections}
+        overrideItemLayout={(layout, item, index) => {
+          layout.size = heights[index];
+        }}
+        onScrollBeginDrag={onScroll}
+        onScrollEndDrag={onScroll}
+        onScroll={debounce(onScroll)}
+        onMomentumScrollBegin={onScroll}
+        onMomentumScrollEnd={onScroll}
+      />
+    </View>
+  );
+};
+
+const SectionList = <T,>(props: Props<T>) => {
+  const selectedItem = useSharedValue('');
+  const sectionNames = props.sections.map((s) => s.name);
+  const visibleIndex = useSharedValue(0);
+  const contentListRef = useRef<ContentList<T>>(null);
+  const tableOfContentsRef = useRef<FlashList<string>>(null);
+
+  return (
+    <View style={styles.container}>
+      <TableOfContents
+        selectedItem={selectedItem}
+        data={sectionNames}
+        visibleIndex={visibleIndex}
+        contentListRef={contentListRef}
+        tableOfContentsRef={tableOfContentsRef}
+      />
+      <Content
+        {...props}
+        visibleIndex={visibleIndex}
+        tableOfContentsRef={tableOfContentsRef}
+        contentListRef={contentListRef}
+      />
+    </View>
+  );
+};
+
+const sectionListStyles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: 'center',
+    backgroundColor: '#111',
+  },
+  flex: {
+    flex: 1,
+  },
+  tableOfContentsElement: { margin: 8, borderRadius: 12, overflow: 'hidden' },
+  sectionHeader: {
+    textAlign: 'center',
+    fontSize: 24,
+    fontWeight: 'bold',
+  },
+  tableOfContents: {
+    top: 0,
+  },
+});
+
+const SCREEN_HEIGHT = Dimensions.get('screen').height;
+
+const sections = Array(Math.floor(Math.random() * 10) + 4)
+  .fill(0)
+  .map((x, i) => ({ name: `Name ${i}`, data: `item ${i}` }));
+
+const getRandomItemHeight = () =>
+  Math.floor(Math.random() * (SCREEN_HEIGHT / 2) + SCREEN_HEIGHT / 3);
+
+const ResizableItem = ({ children }: PropsWithChildren) => {
+  const [height, setHeight] = useState(getRandomItemHeight());
+
+  return (
+    <Pressable onPress={() => setHeight(getRandomItemHeight())}>
+      <View
+        style={{
+          margin: 16,
+          backgroundColor: `#${Math.floor(
+            Math.random() * 8388607 + 8388607
+          ).toString(16)}`,
```

Comment:
what the f*ck?

Reply:
`8388607` nice Mersenne's number, but not even a prime :( Unlucko

Reply:
`FFFFFF/2` 😅